src/app/qgswelcomepageitemsmodel.cpp

```diff
@@ -44,9 +110,8 @@ QVariant QgsWelcomePageItemsModel::data( const QModelIndex& index, int role ) co
   switch ( role )
   {
     case Qt::DisplayRole:
-      return mRecentProjects.at( index.row() ).title;
+      return QString( "<span style='font-size:18px;font-weight:bold;'>%1</span><br>%2" ).arg( mRecentProjects.at( index.row() ).title != mRecentProjects.at( index.row() ).path ? mRecentProjects.at( index.row() ).title : QString( tr( "- untitled -" ) ) ).arg( mRecentProjects.at( index.row() ).path );
```

Comment:
I'd probably add an enum with further roles, starting from UserRole similar to here http://qgis.org/api/classQgsAttributeTableModel.html#ad2a1fab62639c671c9982af13091a67d

http://qgis.org/api/qgsattributetablemodel_8h_source.html#l00052

    enum Role {
      TitleRole = Qt::UserRole,
      PathRole
    };

and add two additional cases.